src/content/reference/react/use-client.md

```diff
@@ -332,13 +333,15 @@ export default function FancyText({title, text}) {
 }
 ```
 
-In this case, it is discouraged to use the `'use client'` directive as it prematurely forces all component usages of `FancyText` to be rendered on the client, which comes at a performance cost. As demonstrated in the earlier Inspirations app example, `FancyText` is used as both a Server or Client Component, depending on where it is imported and used.
+In this case, we don't add the `'use client'` directive, resulting in `FancyText`'s _output_ (rather than its source code) to be sent to the browser when referenced from a Server Component. As demonstrated in the earlier Inspirations app example, `FancyText` is used as both a Server or Client Component, depending on where it is imported and used.
+
+But if `FancyText`'s HTML output was large relative to its source code (including dependencies), it might be more efficient to force it to always be a Client Component. Components that return a long SVG path string are one case where it may be more efficient to force a component to be a Client Component.
 
 ### Using client APIs {/*using-client-apis*/}
 
-Your React app may use client-specific APIs which are dependent on your targeted client. For the browser, some example client APIs include web storage, audio and video manipulation, and device hardware, among [others](https://developer.mozilla.org/en-US/docs/Web/API).
```

Comment:
I think you were trying to be agnostic to web vs native here? but I think it was a bit confusing in practice and better to be clearer

(most of our docs are written with web in mind; I think we probably need a bigger rethink terminology-wise before we could truly update them to be agnostic)